pkg/publicapi/endpoint/requester/endpoints_events.go

```diff
@@ -40,9 +42,13 @@ func (s *Endpoint) events(c echo.Context) error {
 		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
 	}
 
-	legacyEvents := make([]model.JobHistory, len(events))
-	for i := range events {
-		legacyEvents[i] = *legacy.ToLegacyJobHistory(&events[i])
+	legacyEvents := make([]model.JobHistory, 0, len(events))
+	for _, e := range events {
+		if e.Type != models.JobHistoryTypeJobLevel &&
+			e.Type != models.JobHistoryTypeExecutionLevel {
+			continue
+		}
+		legacyEvents = append(legacyEvents, legacy.ToLegacyJobHistory(e))
```

Comment:
We ignore the new history type in the legacy api calls...? I think this is a legacy api, not sure.
Additionally removed gratuitous use of pointers.